calico_containers/calico_ctl/checksystem.py

```diff
@@ -177,3 +181,36 @@ def _check_docker_version(libnetwork=False):
             system_ok = False
 
     return system_ok
+
+def _check_etcd_version():
+    """
+    Check that etcd is running and supported.
+
+    :return: True if etcd version is OK.
+    """
+    system_ok = True
+
+    minimum_version_string = ETCD_VERSION
+    minimum_version = normalize_version(minimum_version_string)
+
+    try:
+        detected_version_string = \
+            client.etcd_client.api_execute("/version", 'GET').data
+    except EtcdConnectionFailed:
+        print >> sys.stderr, "ERROR: Could not connect to etcd at %s" % \
+            etcd_authority
+        system_ok = False
+    else:
+        # Remove everything except numbers and dots from the string.
+        detected_version_string = re.sub(r'[^\d\.]', '',
```

Comment:
Actually, the more I think about this I'm not sure I do like the removing all other characters.

With Semver, the version could be something like 1.2.3-x1 which would get transposed to 1.2.31 with this algorithm.

I think we probably do need to at least handle the - or + separating the X.Y.Z from the pre-release identifier.